components/eam/src/physics/crm/pam/pam_state.h

```diff
@@ -144,7 +153,7 @@ inline void pam_state_set_reference_state( pam::PamCoupler &coupler ) {
   real2d hmean_temp ("hmean_temp" ,nz  ,nens);
   real r_nx_ny  = 1._fp/(nx*ny);  // precompute reciprocal to avoid costly divisions
   // initialize horizontal means
-  parallel_for(SimpleBounds<2>(nz,nens), YAKL_LAMBDA (int k, int iens) {
+  parallel_for(SimpleBounds<2>(nz+1,nens), YAKL_LAMBDA (int k, int iens) {
     hmean_pint(k,iens) = 0;
     if (k < nz) { 
       hmean_pmid (k,iens) = 0;
```

Comment:
This is being set to zero here but not atomically added in the next kernel.

Reply:
yea, good eye, I'll remove this line.

Reply:
fixed.